cogs/respects/respects.py

```diff
@@ -261,5 +261,9 @@ async def payRespects(self, ctx: Context):
                         users = "{}, {}".format(userObj.name, users)
                 message = "**{}** have paid their respects {}".format(users, choice(HEARTS))
 
-            messageObj = await ctx.send(message)
+            messageObj = await ctx.send(
+                message,
+                reference=ctx.message.reference,
```

Comment:
Sorry if it was unclear from the issue, but this looks like it works for the initial reply, but not subsequent invocations within the configured message/time period. Since `ctx.message.reference` uses the command-invoking message, we lose the reply to the original message.

Can we do something like:
- Create `oldRespect` with value `None` at around line 229.
- Somewhere here, check to see if it is not `None`, and if so, use `oldRespect.reference` and plug it into reference, else use `ctx.message.reference`?

Reply:
Feel free to ping me on SAD if you want to chat about it.

Reply:
> Sorry if it was unclear from the issue, but this looks like it works for the initial reply, but not subsequent invocations within the configured message/time period.

Yes this is exactly what I had in mind when I read the last line of #401, though it's also true as you said that it was not very clear when I first read the issue. The situation that prevented me from going down that path was that I imagined using `[p]f` would always result in a message with reference, unless the referenced message is lost (e.g., deleted). But then, respects are paid to a certain something always, lol. Therefore I just made initial working changes and let you review.

> * Create `oldRespect` with value `None` at around line 229.
> * [...] check to see if it is not `None`, and if so, use `oldRespect.reference` and plug it into reference, else use `ctx.message.reference`?

This should clear up everything. I will make the changes.

> Feel free to ping me on SAD if you want to chat about it.

Yep. Will do if something comes up.